inginious-autotest

```diff
@@ -0,0 +1,255 @@
+            l = output1[k]  # in output1, values are lists of size 2
+            t = output2[k]  # in output2, values are tuples of size 2
+            if l[0] != t[0] or l[1] != t[1]:
+                return True
+        return False
+
+    def archive_compare(output1, output2):
+        """
+        Return False because archives are always different
+        """
+        return False
+
+    def generic_compare(output1, output2):
+        """
+        Generic compare for keys
+        :param output1: string
+        :param output2: string
+        :return: see code
+        """
+        return output1 != output2
+
+    func = {
+        "result": result_compare,
+        "problems": problems_compare,
+        "archive": archive_compare
+    }
+    return func.get(key, generic_compare)(output1, output2)
+
+
+def test_task(yaml_data, task, client, client_sync):
+    """
+    Test the task by comparing the new outputs with the old ones
+    :param yaml_data: dict corresponding to the yaml output file for the task
+    :param task: Task object corresponding to the yaml data
+    :param client: backend client of type Client
+    :param client_sync: ClientSync object
+    :return: dict whose the format is specified in compare_all_outputs function doc
+    """
+    if task.get_environment() not in client.get_available_containers():
+        time.sleep(10)
+    if task.get_environment() not in client.get_available_containers():
+        raise Exception('Environment not available')
+    new_output = client_sync.new_job(task, yaml_data['input'])
+    keys = ["result", "grade", "problems", "tests", "custom", "state", "archive", "stdout", "stderr"]
+    old_output = [yaml_data.get(x, None) for x in keys]
+    return compare_all_outputs(old_output, new_output, keys)
+    # tests contains tags
+
+
+def test_web_task(yaml_data, task, res, config):
+    try:
+        web_task = WebAppTask(
+            task.get_course_id(),
+            task.get_id(),
+            yaml_data,
+            task.get_fs(),
+            task.get_translation_fs(),
+            task.get_hook(),
+            config["default_problem_types"],
+            debug=True
+        )
+        web_task.get_context('English')
+    except BaseException as e:
+        res = yaml_data
+
+
+def test_submission_yaml(client, course_factory, path, output, client_sync):
+    # print(os.path.join(test_path, yaml_file.name))
+    with open(path, 'r') as yaml:
+        yaml_data = load(yaml)
+        res = test_task(yaml_data, get_task(yaml_data["courseid"], yaml_data["taskid"], course_factory), client, client_sync)
+        if res != {}:
+            output[path] = res
+
+
+def test_task_yaml(path, output, course_factory, task_name, course_name, config):
+    res = {}
+    with open(path, 'r') as yaml_file:
+        yaml_data = load(yaml_file)
+    test_web_task(yaml_data, get_task(course_name, task_name, course_factory), res, config)
+    if res != {}:
```

Comment:
res is not modified inside `test_web_task`, so this cannot work

Reply:
changed too